shared/src/store/action-reducers/utils/calculate-treatments.ts

```diff
@@ -1,255 +1,334 @@
-            )
-            .map(({ patient }) => patient) ?? [];
 
-    const greenPatients =
-        distancesByStatus.green
-            ?.sort(
-                ({ distance: distanceA }, { distance: distanceB }) =>
-                    distanceA - distanceB
-            )
-            .map(({ patient }) => patient) ?? [];
+    // Patients that got already treated in overrideTreatmentRange should not be treated again in treatmentRange
+    const cateredForPatients = new Set<UUID>();
 
-    for (const patient of redPatients) {
-        if (
-            !caterFor(
-                catering,
+    if (cateringElement.overrideTreatmentRange > 0) {
+        const patientIdsInOverrideRange = SpatialTree.findAllElementsInCircle(
+            state.spatialTrees.patients,
+            cateringElement.position,
+            cateringElement.overrideTreatmentRange
+        );
+        // In the overrideTreatmentRange (the override circle) only the distance to the patient is important - his injuries are ignored
+        for (const patientId of patientIdsInOverrideRange) {
+            caterFor(
+                cateringElement,
                 catersFor,
-                patient,
-                pretriageEnabled,
-                bluePatientsEnabled
-            )
-        ) {
-            break;
+                getElement(state, 'patients', patientId),
+                state.configuration.pretriageEnabled,
+                state.configuration.bluePatientsEnabled
+            );
+            cateredForPatients.add(patientId);
         }
     }
 
-    for (const patient of yellowPatients) {
-        if (
-            !caterFor(
-                catering,
-                catersFor,
-                patient,
-                pretriageEnabled,
-                bluePatientsEnabled
-            )
-        ) {
-            break;
-        }
+    if (
+        cateringElement.treatmentRange <= 0 ||
+        // Only look for more patients to treat, if there still is capacity (performance improvement)
+        (!couldCaterFor('red', cateringElement, catersFor) &&
+            !couldCaterFor('yellow', cateringElement, catersFor) &&
+            !couldCaterFor('green', cateringElement, catersFor))
+    ) {
+        return;
     }
 
-    for (const patient of greenPatients) {
-        if (
-            !caterFor(
-                catering,
-                catersFor,
-                patient,
-                pretriageEnabled,
-                bluePatientsEnabled
+    const patientsInTreatmentRange: Mutable<Patient>[] =
+        SpatialTree.findAllElementsInCircle(
+            state.spatialTrees.patients,
+            cateringElement.position,
+            cateringElement.treatmentRange
+        )
+            // Filter out every patient in the overrideTreatmentRange
+            .filter(
+                (patientId) =>
+                    // TODO: removing the ! makes the code break, but it should just recalculate for every patient again?!
```

Comment:
@Dassderdie Do you have an idea why this could have happened? Maybe I got this wrong that this happens

Reply:
`makes the code break` is not a very specific description.

What removing the `!` does do, is effectively reduce the `treatmentRange` to the `overrideTreatmentRange` because only patients treated in `overrideTreatmentRange` will be looked at in the next calculation steps.